skopt/space/space.py

```diff
@@ -550,6 +552,49 @@ def is_real(self):
         """
         return all([isinstance(dim, Real) for dim in self.dimensions])
 
+    @classmethod
+    def from_yaml(cls, yml_path, namespace=None):
+        """Create Space from yaml configuration file
+
+        Parameters
+        ----------
+        * `yml_path` [str]:
+           Full path to yaml configuration file
+        * `namespace` [str, default=None]:
+           Namespace within configuration file to use
+
+        Returns
+        -------
+        * `space` [Space]:
+           Instantiated Space object
+        """
+        with open(yml_path, 'rb') as f:
+            cfg = yaml.load(f)
+
+        this_module = sys.modules[__name__]
+        # Dynamically create list of subclass names for Dimension
+        dimension_classes = ['Real', 'Integer', 'Categorical']
```

Comment:
I'd make a dictionary like `{'real': Real, 'categorical': Categorical, ...}` to take care of converting from the yaml strings to classes. Purely because `this_module` and `getattr` looks like there will be some subtle edge cases where it stops working that we will run into at some point in the future (but that is just a feeling).